src/base/compute/owl_computation_cpu_eval.ml

```diff
@@ -19,232 +19,241 @@ module Make
 
   (* utility functions *)
 
-  let update_validity x =
-    validate x;
-    Array.iter invalidate (get_vnode x)
+  let invalidate_opt x_option =
+    match x_option with
+    | Some x -> invalidate x
+    | None   -> ()
 
```

Comment:
`let invalidate_opt = function Some x -> invalidate x | None -> ()`, just a shorter version if you need.